### PR TITLE
Simlify with matches!()

### PR DESCRIPTION
**Pull request by Veetaha**

### CODE REVIEW

File: crates/ra_assists/src/handlers/fix_visibility.rs [hunk 1]
```diff
@@ -179,11 +179,7 @@ fn target_data_for_def(
...
     node.children_with_tokens()
-        .skip_while(|it| match it.kind() {
-            WHITESPACE | COMMENT | ATTR => true,
-            _ => false,
-        })
-        .next()
+        .find(|it| !matches!(it.kind(), WHITESPACE | COMMENT | ATTR))
         .map(|it| it.text_range().start())
         .unwrap_or_else(|| node.text_range().start())
 }
```
Comment (matklad) [inline crates/ra_assists/src/handlers/fix_visibility.rs:185]: Let’s pull both fidenticsl unctions up to until

File: crates/ra_assists/src/handlers/merge_match_arms.rs [hunk 2]
```diff
@@ -81,10 +81,7 @@ pub(crate) fn merge_match_arms(acc: &mut Assists, ctx: &AssistContext) -> Option
 }
 
 fn contains_placeholder(a: &ast::MatchArm) -> bool {
-    match a.pat() {
-        Some(ra_syntax::ast::Pat::PlaceholderPat(..)) => true,
-        _ => false,
-    }
+    matches!(a.pat(), Some(ra_syntax::ast::Pat::PlaceholderPat(..)))
```
Comment (matklad) [inline crates/ra_assists/src/handlers/merge_match_arms.rs:84]: Let’s import ast while we are at it.

File: crates/ra_hir_def/src/nameres.rs [hunk 3]
```diff
@@ -137,10 +137,7 @@ impl ModuleOrigin {
     }
 
     pub fn is_inline(&self) -> bool {
-        match self {
-            ModuleOrigin::Inline { .. } => true,
-            ModuleOrigin::CrateRoot { .. } | ModuleOrigin::File { .. } => false,
-        }
+        matches!(self, ModuleOrigin::Inline { .. })
```
Comment (matklad) [inline crates/ra_hir_def/src/nameres.rs:140]: Note that original pattern is exhaustive, while the new one is not. The first optio is better, as it is more resilient to changes.

File: crates/ra_assists/src/handlers/fix_visibility.rs (hunk 1 above)
Comment (Veetaha) [reply crates/ra_assists/src/handlers/fix_visibility.rs]: Done
https://github.com/rust-analyzer/rust-analyzer/pull/5105/commits/503f9498cd469c6467250c02bd8f6dbce973dd57

File: crates/ra_assists/src/handlers/merge_match_arms.rs (hunk 2 above)
Comment (Veetaha) [reply crates/ra_assists/src/handlers/merge_match_arms.rs]: Done https://github.com/rust-analyzer/rust-analyzer/pull/5105/commits/503f9498cd469c6467250c02bd8f6dbce973dd57

File: crates/ra_hir_def/src/nameres.rs (hunk 3 above)
Comment (Veetaha) [reply crates/ra_hir_def/src/nameres.rs]: Done https://github.com/rust-analyzer/rust-analyzer/pull/5105/commits/503f9498cd469c6467250c02bd8f6dbce973dd57

**Pull request merged**